sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -136,6 +137,40 @@ export class Recorder {
     }
   }
 
+  /**
+   * revertRequestChanges reverts the request in record and playback modes back to the existing url.
+   * Works for both core-v1 and core-v2
+   *
+   * - WebResource -> core-v1
+   * - PipelineRequest -> core-v2
+   *
+   * Wrokflow:
+   *   1. recorderHttpPolicy calls this method after the request is made
+   *   2. "redirectRequest" method is called to update the request with the proxy-tool url
+   *   3. Request hits the proxy tool, proxy-tool hits the service and returns the response
+   *   4. Using `revertRequestChanges`, we revert the request back to the original url
+   */
+  private revertRequestChanges(request: WebResource | PipelineRequest): void {
+    logger.info(
+      `[Recorder#revertRequestChanges] "undo"s the URL changes made by the recorder to hit the test proxy after the response is received,`,
+      request
+    );
+    const headers = request.headers;
+    if (isHttpHeadersLike(headers)) {
+      // core-v1
+      headers.remove("x-recording-id");
+      headers.remove("x-recording-mode");
+    } else {
+      // core-v2
+      headers.delete("x-recording-id");
+      headers.delete("x-recording-mode");
+    }
+    request.url = request.url.replace(
```

Comment:
I wonder if we can just store the previous request URL somewhere and place it back using that instead of doing the string replacement? Don't know if there's much practical difference.

Reply:
sounds safer.

Reply:
We can't store the url in the recorder, because we might be making parallel requests to different endpoints and it can potentially break.
We must rely on the `"x-recording-upstream-base-uri` I think.

Makes sense?

Reply:
I think mapping the URLs is too much and if we'd have to do that then let's just go with what you have. But can't we just do:

```ts
const originalUrl = request.url;
this.redirectRequest(request);
const response = await next(request);
this.handleTestProxyErrors(response);
this.revertRequestChanges(request, originalUrl); // pass originalUrl into `revertRequestChanges` and update it there
```

That avoids making the recorder class stateful. Honestly though, not too fussed -- I can't think of any situations when the current approach wouldn't work.

Reply:
This sounds good, with less maintenance. Updating with your suggestion.